docs/ref/models/lookups.txt

```diff
@@ -83,7 +83,7 @@ field references, aggregates, and ``Transform`` are examples that follow this
 API. A class is said to follow the query expression API when it implements the
 following methods:
 
-.. method:: as_sql(self, compiler, connection)
+.. method:: as_sql(self, compiler, connection, **kwargs)
```

Comment:
And call out that `**kwargs` will be mixed with `self.extra` for the purposes of interpolating into `self.template`. I'd also callout that `template` kwarg is a special case which can be used to override `self.template` for that particular invocation.